docs/roadmap.md

```diff
@@ -7,15 +7,108 @@ has_children: false
 ---
 
 # Roadmap
+{: .no_toc }
 
-- [ ] **General**{: .label } Improved Monitoring
-- [ ] **Block Adapter**{: .label } Additional storage adapters (Azure storage?)
-- [ ] **Integrations**{: .label } Native Hadoop/Spark Filesystem & OutputCommitter
-- [ ] **Integrations**{: .label } Java SDK
-- [ ] **Integrations**{: .label } Airflow Operator
-- [ ] **Auth**{: .label } Support instance Profiles
-- [ ] **Auth**{: .label } IAM/KMS integration
-- [ ] **CI**{: .label } Web Hook support
-- [ ] **CI**{: .label } Protected branches
-- [ ] **UI**{: .label } Better pagination and data exploration
+## Table of contents
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+
+## Architecture
+
+_TL;DR_ - After receiving feedback on early versions of lakeFS, project **["lakeFS on the Rocks"](https://docs.google.com/document/d/1jzD7-jun-tdU5BGapmnMBe9ovSzBvTNjXCcVztV07A4/edit?usp=sharing){:target="_blank"}** represents a set of changes to the architecture and data model of lakeFS. The main motivators are simplicity, reduced barriers of entry, scalability -  and the added benefit of having lakeFS adhere more closely to Git in semantics and UX.
+{: .note .note-warning }
+
+There are 3 big shifts in design:
+
+
+1. Using the underlying object store as a source of truth for all committed data. We do this by storing commits as RocksDB SSTable files, where each commit is a “snapshot” of a given repository, split across multiple SSTable files that could be reused across commits.
+1. Removal of PostgreSQL as a dependency: scaling it for very high throughput while keeping it predictable in performance for different loads and access patterns has a very high operational cost.
+1. Separating out the metadata operations into a separate service (with the S3 API Gateway remaining as a stateless client for this service). Would allow for the development of “native” clients for big data tools that don’t require passing the data itself through lakeFS, but rather talk directly to the underlying object store.
```

Comment:
```suggestion
1. Extract metadata operations into a separate service (with the S3 API Gateway remaining as a stateless client for this service). Would allow for the development of “native” clients for big data tools that don’t require passing the data itself through lakeFS, but rather talk directly to the underlying object store.
```